src/main/java/org/broadinstitute/hellbender/engine/LocusWalker.java

```diff
@@ -0,0 +1,154 @@
+ * variants/Features.
+ *
+ * LocusWalker authors must implement the apply() method to process each position, and may optionally implement
+ * onTraversalStart() and/or onTraversalDone().
+ *
+ * @author Daniel Gómez-Sánchez (magicDGS)
+ */
+public abstract class LocusWalker extends GATKTool {
+
+    @Argument(fullName = "disable_all_read_filters", shortName = "f", doc = "Disable all read filters", common = false, optional = true)
+    public boolean disableAllReadFilters = false;
+
+    /**
+     * Should the LIBS keep unique reads?
+     */
+    protected boolean KEEP_UNIQUE_READ_LIST_IN_LIBS = false;
+
+    /**
+     * LocusWalkers requires read sources
+     */
+    @Override
+    public boolean requiresReads() {
+        return true;
+    }
+
+    /**
+     * Does this tool require deletions in the AlignmentContext? Tools that do should override to return true.
+     *
+     * @return {@code true} if this tool requires deletions, {@code false} otherwise
+     */
+    public boolean includeDeletions() {
+        return true;
+    }
+
+    /**
+     * Returns the read filter (simple or composite) that will be applied to the reads in each window.
+     *
+     * The default implementation uses the {@link org.broadinstitute.hellbender.engine.filters.WellformedReadFilter} filter with all default options,
+     * as well as the {@link ReadFilterLibrary#MAPPED} filter.
+     *
+     * Default implementation of {@link #traverse()} calls this method once before iterating
+     * over the reads and reuses the filter object to avoid object allocation. Nevertheless, keeping state in filter objects is strongly discouraged.
+     *
+     * Subclasses can override to provide their own filters
+     * Multiple filters can be composed by using {@link org.broadinstitute.hellbender.engine.filters.ReadFilter} composition methods.
+     */
+    public CountingReadFilter makeReadFilter(){
+        return new CountingReadFilter("Wellformed", new WellformedReadFilter(getHeaderForReads()))
+                .and(new CountingReadFilter("Mapped", ReadFilterLibrary.MAPPED));
+    }
+
+    /**
+     * Get the information about how to downsample the reads. By default {@link org.broadinstitute.hellbender.utils.downsampling.DownsamplingMethod#NONE} is returned.
+     * Subclasses should override it to provide own downsampling methods.
+     *
+     * @return the downsampling method for the reads
+     */
+    public DownsamplingMethod getDownsamplingMethod() {
+        // TODO: change when downsampling command line options are implemented
+        return DownsamplingMethod.NONE;
+    }
+
+    /**
+     * Marked final so that tool authors don't override it. Tool authors should override onTraversalStart() instead.
+     */
+    @Override
+    protected final void onStartup() {
+        super.onStartup();
+        if ( hasIntervals() ) {
+            reads.setIntervalsForTraversal(intervalsForTraversal);
```

Comment:
This should change to `reads.setTraversalBounds(intervalArgumentCollection.getTraversalParameters(getHeaderForReads().getSequenceDictionary()))` once you rebase onto the latest master.